data-entry/index.html.in

```diff
@@ -80,9 +80,9 @@
                                     <!-- dropdown select input -->
                                     <div ng-switch-when="dropdown">
                                         <ui-select ng-model="form.dataEntryModel.rows[rowIndex][column.name]" theme="select2" ng-disabled="disabled">
-                                            <ui-select-match name="{{::column.name}}" placeholder="Select a value">{{$select.selected | underscoreToSpace | toTitleCase}}</ui-select-match>
-                                            <ui-select-choices repeat="domainValue in form.dataEntryModel.domainValues[column.name] | filter: $select.search" >
-                                                <div ng-bind-html="domainValue | underscoreToSpace | toTitleCase | highlight: $select.search"></div>
+                                            <ui-select-match name="{{::column.name}}" placeholder="Select a value">{{$select.selected.display | underscoreToSpace | toTitleCase}}</ui-select-match>
+                                            <ui-select-choices repeat="domainValue.key as domainValue in form.dataEntryModel.domainValues[column.name] | filter: $select.search" >
+                                                <div ng-bind-html="domainValue.display | underscoreToSpace | toTitleCase | highlight: $select.search"></div>
```

Comment:
`underscoreToSpace` and `toTitleCase` heuristics only applied to "display names" and never apply to value.

In addition, we are getting rid of this (default) transformation on our column names and replacing with the explicitly defined `Column.displayname` so that the presentation code should not be making these decisions in the future.